Azure/azure-cli
# [RDBMS] `az postgres flexible-server geo-restore/replica`: Introduce read replicas and geo-restore

**Related command**
- `az postgres flexible-server geo-restore`
- `az postgres flexible-server replica create`
- `az postgres flexible-server replica list`
- `az postgres flexible-server replica stop-replication`

**Description**
Introduce replicas and geo-restore for PostgreSQL flexible server:
- Replica location can be different from primary server location.
- Restored and geo-restored servers can decide whether to be geo-redundant or not.
- From a server with public access, we can only:
   - create a replica with public access.
   - geo-restore with public access
- From a server with vnet, we can only:
   - create a replica with a vnet in same region.
   - geo-restore with a vnet in paired region.

Changes in code:
- Add `geo_redundant_backup` argument in functions `flexible_server_create` and `flexible_server_restore`.
- Add functions `flexible_server_georestore`, `flexible_replica_create`, `flexible_replica_stop`.
- Refactor vnet, subnet and private dns zone provisionings in method `flexible_server_provision_network_resource`.
- Update azure-mgmt-rdbms to version 10.2.0b5.

**Testing Guide**

See examples in _helptext_pg.py and updated tests.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
